src/Front/Pattern/Desugar.hs

```diff
@@ -79,11 +79,27 @@ desugarTopDefs = mapM desugarTopDef
 desugarDefs :: [A.Definition] -> DesugarFn [A.Definition]
 desugarDefs = mapM desugarDef
 
--- WARN: only body is desugared
 desugarDef :: A.Definition -> DesugarFn A.Definition
-desugarDef (A.DefFn i ps t e) = A.DefFn i ps t <$> desugarExpr e
+desugarDef (A.DefFn i [] t e) = return (A.DefFn i [] t e)
+desugarDef (A.DefFn i ps t e) =
+  let (pats, rese) = helper ps e (0::Int) in
```

Comment:
I would put `(pats, rese) = helper ps e (0::Int)` inside the where clause and get rid of this let.